linkingExamples_test.go

```diff
@@ -0,0 +1,133 @@
+	// If you initialize a custom LinkSystem, you can control this more directly;
+	//  these registry systems are only here as defaults.
+
+	// Now: time to apply the LinkSystem, and do the actual store operation!
+	lnk, err := lsys.Store(
+		ipld.LinkContext{}, // The zero value is fine.  Configure it it you want cancellability or other features.
+		lp,                 // The LinkPrototype says what codec and hashing to use.
+		n,                  // And here's our data.
+	)
+	if err != nil {
+		panic(err)
+	}
+
+	// That's it!  We got a link.
+	fmt.Printf("link: %s\n", lnk)
+	fmt.Printf("concrete type: `%T`\n", lnk)
+
+	// Remember: the serialized data was also stored to the 'store' variable as a side-effect.
+	//  (We set this up back when we customized the LinkSystem.)
+	//  We'll pick this data back up again in the example for loading.
+
+	// Output:
+	// link: bafyrkmbukvrgzcs6qlsh4wvkvbe5wp7sclcblfnapnb2xfznisbykpbnlocet2qzley3cpxofoxqrnqgm3ta
+	// concrete type: `cidlink.Link`
+}
+
+func ExampleLoadingLink() {
+	// Let's say we want to load this link (it's the same one we just created in the example above).
+	cid, _ := cid.Decode("bafyrkmbukvrgzcs6qlsh4wvkvbe5wp7sclcblfnapnb2xfznisbykpbnlocet2qzley3cpxofoxqrnqgm3ta")
+	lnk := cidlink.Link{cid}
+
+	// Let's get a LinkSystem.  We're going to be working with CID links,
+	//  so let's get the default LinkSystem that's ready to work with those.
+	// (This is the same as we did in ExampleStoringLink.)
+	lsys := cidlink.DefaultLinkSystem()
+
+	// We need somewhere to go looking for any of the data we might want to load!
+	//  We'll use an in-memory store for this.  (It's a package scoped variable.)
+	//   (This particular memory store was filled with the data we'll load earlier, during ExampleStoringLink.)
+	//  You can use any kind of storage system here;
+	//   you just need a function that conforms to the ipld.BlockReadOpener interface.
+	lsys.StorageReadOpener = (&store).OpenRead
+
+	// We'll need to decide what in-memory implementation of ipld.Node we want to use.
+	//  Here, we'll use the "basicnode" implementation.  This is a good getting-started choice.
+	//   But you could also use other implementations, or even a code-generated type with special features!
+	np := basicnode.Prototype.Any
+
+	// Before we use the LinkService, NOTE:
+	//  There's a side-effecting import at the top of the file.  It's for the dag-cbor codec.
+	//  See the comments in ExampleStoringLink for more discussion of this and why it's important.
+
+	// Apply the LinkSystem, and ask it to load our link!
+	n, err := lsys.Load(
+		ipld.LinkContext{}, // The zero value is fine.  Configure it it you want cancellability or other features.
+		lnk,                // The Link we want to load!
+		np,                 // The NodePrototype says what kind of Node we want as a result.
+	)
+	if err != nil {
+		panic(err)
```

Comment:
you declared Must methods, so you could also use them here to keep the example shorter